nixos/modules/services/games/crossfire-server.nix

```diff
@@ -0,0 +1,119 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.crossfire-server;
+  serverPort = 13327;
+in {
+  options.services.crossfire-server = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        If enabled, the Crossfire game server will be started at boot.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.crossfire-server-stable;
+      defaultText = "pkgs.crossfire-server-stable";
+      example = literalExample "pkgs.crossfire-server-latest";
+      description = ''
+        The package to use for the Crossfire server (and map/arch data, if you
+        don't change dataDir).
+      '';
+    };
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "${cfg.package}/share/crossfire";
+      description = ''
+        Where to load readonly data from -- maps, archetypes, treasure tables,
+        and the like. If you plan to edit the data on the live server (rather
+        than overlaying the crossfire-maps and crossfire-arch packages and
+        nixos-rebuilding), point this somewhere read-write and copy the data
+        there before starting the server.
+      '';
+    };
+
+    runtimeDir = mkOption {
+      type = types.str;
+      default = "/srv/crossfire";
+      description = ''
+        Where to store runtime data (save files, persistent items, etc).
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to open ports in the firewall for the server.
+      '';
+    };
+
+    etc = mkOption {
+      type = types.attrsOf types.str;
+      description = ''
+        Text to append to the corresponding configuration files. For the format
+        of each of these, consult the documentation in the corresponding file
+        in ${cfg.package}/etc/.
```

Comment:
Avoid `${cfg.package}` here to avoid manual rebuilds if the package is changes.

Reply:
What would you recommend instead? If I point them at `/etc/crossfire` they don't have the docs  until the package is enabled; there's no good online documentation to point them at at; and it seems a bit mean to tell them to rummage around "somewhere in `/nix/store`" for it.

Reply:
Maybe online documentation or something like: 

```
cd $(nix-build '<nixpkgs>' -A crossfire --no-out-link)/etc
```